src/Tribe/Views/V2/Views/By_Day_View.php

```diff
@@ -187,13 +187,24 @@ public function get_grid_days( $date = null, $force = false ) {
 			$last_grid_day  = $days->end;
 			$end            = tribe_end_of_day( $last_grid_day->format( Dates::DBDATETIMEFORMAT ) );
 
-			$view_event_ids = tribe_events()
+			/*
+			 * Sort events in duration ascending order to make sure events that start on the same date and time
+			 * will be correctly positioned for multi-day, or all-day, parsing.
+			 * If not explicit, then events with the same start date and time would be sorted in the order MySQL
+			 * read them (not guaranteed).
+			 */
+			$order_by                   = tribe_normalize_orderby( $order_by, $order );
+			$order_by['event_duration'] = 'ASC';
+
+			$repo           = tribe_events()
```

Comment:
:warning: Equals sign not aligned correctly; expected 1 space but found 11 spaces

<sub>hash: f6b249f865f0959614cdc7c0b145fe4c</sub>